### PR TITLE
D4CHAT-41 feat: Implement user service

### PR DESCRIPTION
**Pull request by GrigoriiBerezin**

### CODE REVIEW

File: src/main/scala/ru/d4team/chat/services/PersonService.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  val live: URLayer[PersonDAO, PersonService] = ZLayer.fromFunction(new PersonServiceImpl(_))
+}
+
+trait PersonService {
+  def addPerson(person: Person): Task[Person]
+  def findPerson(id: UUID): Task[Option[Person]]
+  def getAll: Task[List[Person]]
+}
+
+final class PersonServiceImpl(dao: PersonDAO) extends PersonService {
```
Comment (AlekseiLitkovetc) [inline src/main/scala/ru/d4team/chat/services/PersonService.scala:19]: Consider using a case class

For more information, see https://zio.dev/reference/service-pattern/#2-service-implementation

**Review by AlekseiLitkovetc — CHANGES_REQUESTED** [review]

File: src/main/scala/ru/d4team/chat/services/PersonService.scala (hunk 1 above)
Comment (GrigoriiBerezin) [reply src/main/scala/ru/d4team/chat/services/PersonService.scala:19]: Don't go away, I'll fix it right now :)

File: src/main/scala/ru/d4team/chat/services/PersonService.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package ru.d4team.chat.services
+
+import ru.d4team.chat.dao.PersonDAO
+import ru.d4team.chat.models.person.Person
+import zio._
+
+import java.util.UUID
+
+object PersonService {
+  val live: URLayer[PersonDAO, PersonService] = ZLayer.fromFunction(new PersonServiceImpl(_))
```
Comment (AlekseiLitkovetc) [inline src/main/scala/ru/d4team/chat/services/PersonService.scala:10]: As we use case class, let's use a little bit different style:
```scala
ZLayer.fromFunction(PersonServiceImpl.apply _)
```

File: src/main/scala/ru/d4team/chat/services/PersonService.scala [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+final case class PersonServiceImpl(dao: PersonDAO) extends PersonService {
+  override def addPerson(person: Person): Task[Person] = dao.addPerson(person)
+
+  override def findPerson(id: UUID): Task[Option[Person]] =
+    dao.findPerson(id)
+
+  override def getAll: Task[List[Person]] = dao.getPersons
+}
```
Comment (AlekseiLitkovetc) [inline src/main/scala/ru/d4team/chat/services/PersonService.scala:26]: Also, to be consistent with other services, let's use this order:
1. contract
2. object
3. implementation

**Review by AlekseiLitkovetc — CHANGES_REQUESTED** [review]

**Review by AlekseiLitkovetc — APPROVED** [review]

**Review by PatokovN — APPROVED** [review]

**Pull request merged**